src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -608,81 +608,85 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
 
-                          errorHandler:^(NSError *) {
-                              // Not much we can do here
-                          }];
-    };
+                      Commands::QueryImageResponse::Type delegateResponse;
+                      ConvertFromQueryImageResponseParms(data, delegateResponse);
+
+                      // If BDX protocol is not supported, send a response with status - Protocol Not supported and reset state.
+                      if (![commandParams.protocolsSupported
+                              containsObject:@(MTROtaSoftwareUpdateProviderOTADownloadProtocolBDXSynchronous)]) {
+                          Commands::QueryImageResponse::Type protocolNotSupportedresponse;
+                          protocolNotSupportedresponse.status
+                              = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                          handler->AddResponse(cachedCommandPath, protocolNotSupportedresponse);
+                          handle.Release();
+                          gOtaSender.ResetState();
+                      } else if (![data.status isEqual:@(MTROtaSoftwareUpdateProviderOTAQueryStatusUpdateAvailable)]) {
+
+                          // If BDX protocol is supported but update is not available, send the delegate response and reset state
+                          handler->AddResponse(cachedCommandPath, delegateResponse);
+                          handle.Release();
+                          gOtaSender.ResetState();
+                      } else {
+
+                          // If there is an update available, return a response with status - Available and image uri set in the
+                          // delegate response
+                          auto fabricIndex = handler->GetSubjectDescriptor().fabricIndex;
+                          auto nodeId = handler->GetSubjectDescriptor().subject;
+                          CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
+                          if (CHIP_NO_ERROR != err) {
+                              LogErrorOnFailure(err);
+
+                              // Handle busy error separately as we have a query image response status that maps to busy
+                              if (err == CHIP_ERROR_BUSY) {
+                                  Commands::QueryImageResponse::Type busyResponse;
+                                  busyResponse.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
+                                  busyResponse.delayedActionTime.SetValue(
+                                      delegateResponse.delayedActionTime.ValueOr(kDelayedActionTimeSeconds));
+                                  handler->AddResponse(cachedCommandPath, busyResponse);
+                                  handle.Release();
+                                  return;
+                              } else {
```

Comment:
Else after return will fail lints.

I think this whole block might be a bit more readable as:
```
  CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
  if (err == CHIP_ERROR_BUSY) {
    // Handle busy error separately as we have a query image response status that maps to busy
    // Maybe explicitly log we are busy?
    ...
    return;
  }

  if (err != CHIP_NO_ERROR) {
    // Some other failure
    LogErrorOnFailure(err);
    ...
    return;
  }
```

so things are less indented.

Reply:
done